FitGoal/FitGoal/ExerciseDetailViewController/DetailViewController.swift

```diff
@@ -0,0 +1,298 @@
+        gradientView.startPoint = CGPoint(x: 0, y: 0.70)
+        gradientView.endPoint = CGPoint(x: 1, y: 0.74)
+        gradientView.alpha = 0.15
+        return gradientView
+    }()
+    
+    //MARK: -VC life cycle
+    
+    init(exercise: Exercise) {
+        self.imageURL = exercise.url
+        self.cellTitle.attributedText = exercise.name.uppercased().formattedText(
+            font: "Oswald-Medium",
+            size: 34,
+            color: .white,
+            kern: -0.14
+        )
+        self.text.attributedText = exercise.description.formattedText(
+            font: "Roboto-Light",
+            size: 15,
+            color: #colorLiteral(red: 0.52, green: 0.53, blue: 0.57, alpha: 1),
+            kern: 0.3,
+            lineSpacing: 6
+        )
+        super.init(nibName: nil, bundle: nil)
+        fetchImage()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = #colorLiteral(red: 0.9647058824, green: 0.9647058824, blue: 0.9647058824, alpha: 1)
+        view.addSubview(scrollView)
+        view.addSubview(startExerciseButton)
+        grayShadow.addSubview(exerciseImage)
+        
+        
+        let views = [gradientBackgroundView,
+                     cellTitle,
+                     shadowWithGradient,
+                     grayShadow,
+                     imageGradient,
+                     playButton,
+                     descriptionView,
+                     descriptionTitle,
+                     text
+        ]
+        add(subviews: views, to: scrollView)
+        setConstraints()
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        scrollView.frame = CGRect(x: 0, y: 0, width: view.bounds.width, height: view.bounds.height - 100)
+        scrollView.layoutIfNeeded()
+        
+        exerciseImage.frame = grayShadow.bounds
+        imageGradient.frame = grayShadow.frame
+        playButton.center = grayShadow.center
+        
+        startExerciseButton.frame = CGRect (x: 16, y: view.bounds.maxY - 100, width: view.bounds.width - 32, height: 54)
+        startExerciseButton.layer.cornerRadius = startExerciseButton.bounds.height/2
+    }
+    
+    private func setConstraints(){
+        setTitleLabelConstraints()
+        setGrayShadowConstraints()
+        setShadowWithGradientConstraints()
+        setDescriptionViewConstraints()
+        setTextConstraints()
+        setLabelConstraints()
+    }
+    
+    private func add(subviews: [UIView],  to parentView: UIView) {
+        subviews.forEach { view in
+            parentView.addSubview(view)
+        }
+    }
+    
+    private func fetchImage() {
+        imageURL.fetchImage { result in
+            DispatchQueue.main.async {
+                guard self.imageURL == self.imageURL else { return }
```

Comment:
This comparison will always be true because you are comparing the same values. Remember, the comparison with the current `imageURL` was needed in other places because the cells were reused, but here you don't need to do that.

Other thing, to avoid having the `imageURL` field in the class when there is no usage for it, it would be better to pass it as a parameter to this `fetchImage` function instead. After that you can remove that field.